File: 2021/April/Day 21/monotonic-array.py
# source : https://leetcode.com/problems/monotonic-array/submissions/
class Solution(object):
    def isMonotonic(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """

        inc = 0
        dec =0


        prev = A[0];

        for i in range(1,len(A)):
            if(A[i]>prev):
                inc = 1;
            elif(A[i]<prev):
                dec = 1;

            if(inc and dec):
                return False
            prev = A[i]
        if( not (inc) and dec) or (not(dec) and inc) or (not(inc) and not(dec)):
            return True
        return False

File: 2021/August/Day 18/monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {

        int inc = 0;
        int dec = 0;
        for(int i=1;i<size(nums);i++)
        {

            if(nums[i]>nums[i-1])
            {
                inc =1;
            }
            else if(nums[i]<nums[i-1])
            {
                dec =1;
            }

            if(inc==1 && dec==1)
            {
                return false;
            }
        }
        return true;

    }
};
